x-pack/legacy/plugins/uptime/common/runtime_types/ping/ping.ts

```diff
@@ -0,0 +1,169 @@
+export const PingType = t.intersection([
+  t.type({
+    '@timestamp': t.string,
+    monitor: t.intersection([
+      t.type({
+        duration: t.type({
+          us: t.number,
+        }),
+        id: t.string,
+        status: t.string,
+        type: t.string,
+      }),
+      t.partial({
+        check_group: t.string,
+        ip: t.string,
+        name: t.string,
+        timespan: t.partial({
+          gte: t.string,
+          lte: t.string,
+        }),
+      }),
+    ]),
+  }),
+  t.partial({
+    agent: t.intersection([
+      t.type({
+        ephemeral_id: t.string,
+        hostname: t.string,
+        id: t.string,
+        type: t.string,
+        version: t.string,
+      }),
+      t.partial({
+        name: t.string,
+      }),
+    ]),
+    container: t.partial({
+      id: t.string,
+      image: t.partial({
+        name: t.string,
+        tag: t.string,
+      }),
+      name: t.string,
+      runtime: t.string,
+    }),
+    ecs: t.partial({
+      version: t.string,
+    }),
+    error: t.intersection([
+      t.partial({
+        code: t.string,
+        id: t.string,
+        stack_trace: t.string,
+        type: t.string,
+      }),
+      t.type({
+        // this is _always_ on the error field
+        message: t.string,
+      }),
+    ]),
+    http: t.partial({
+      request: t.partial({
+        body: t.partial({
+          bytes: t.number,
+          content: t.partial({
+            text: t.string,
+          }),
+        }),
+        bytes: t.number,
+        method: t.string,
+        referrer: t.string,
+      }),
+      response: t.partial({
+        body: HttpResponseBodyType,
+        bytes: t.number,
+        redirects: t.string,
+        status_code: t.number,
+      }),
+      version: t.string,
+    }),
+    icmp: t.partial({
+      requests: t.number,
+      rtt: t.partial({
+        us: t.number,
+      }),
+    }),
+    kubernetes: t.partial({
+      pod: t.partial({
+        name: t.string,
+        uid: t.string,
+      }),
+    }),
+    observer: t.partial({
+      geo: t.partial({
+        name: t.string,
+      }),
+    }),
+    resolve: t.partial({
+      ip: t.string,
+      rtt: t.partial({
+        us: t.number,
+      }),
+    }),
+    summary: t.partial({
+      down: t.number,
+      up: t.number,
+    }),
+    tags: t.string,
+    tcp: t.partial({
+      rtt: t.partial({
+        connect: t.partial({
+          us: t.number,
+        }),
+      }),
+    }),
+    tls: TlsType,
+    // should this be partial?
+    url: t.partial({
+      domain: t.string,
+      full: t.string,
+      port: t.number,
+      scheme: t.string,
+    }),
+  }),
+]);
+
+export type Ping = t.TypeOf<typeof PingType>;
+
+export const PingsResponseType = t.type({
+  total: t.number,
+  locations: t.array(t.string),
+  pings: t.array(PingType),
+});
+
+export type PingsResponse = t.TypeOf<typeof PingsResponseType>;
+
+export interface GetPingsParams {
+  dateRangeStart: string;
+  dateRangeEnd: string;
```

Comment:
i think dateRangeStart and dateRangeEnd are too verbose, we should make an effort to make it bit simple, either go for an approach of considering it as something like 
`range: {to:, from:}` or just use dateStart and dateEnd.

Reply:
I agree, I've been using `to` and `from` in some places. I will modify this one.

Reply:
I added a `DateRange` type to the common `runtime_types` and I'm using that now in this type. https://github.com/elastic/kibana/pull/59392/commits/1df28d66e4cf2e0a6217cca8d37ed8b3dfbcf2da